### DISCUSSION TITLE
Calling Function from Maya

### DISCUSSION THREAD

**Discussion by deepakxyz** [nxt-dev/nxt_editor#129] [Q&A] [answered]
How do I call a function which is declared on one node from an anther node? or a function which is already being declared on the Maya terminal.

MayaCodeEditor or NXT Node1 >
```python
def test():
    print('Test function')
```
NXT Node2 >
```python
test()
```

Is this possible?

**Comment by ImLucasBrown** [comment, marked as answer]
There are a couple of approaches you could use.

1. Put your function in the World Node's code block. Function, variables, modules, ect defined in the world node are available globally. You can access the world node by double clicking the layer's name or by pressing `/` and `return` (by default, `/` brings up the node search box).  

2. You could declare the function as a global. `global test`. The global keyword allows all nodes to access the name.

3. Callables can be stored in the `STAGE` object. So after defining the function you could do something like `STAGE.test = test`, then call it in a later node via `STAGE.test()`.

All of these are valid ways to work, however personally I tend to use NXT to formulate function calls and orchestrate the sequence of events. So in my case I would import `test` from an external library, but it all depends on the use case and how hard you want to buy into NXT as the home for your code.

**Reply by sgodanas** [reply]
Thank you for the tips. I tried it, but '/' followed by return does not activate the layer.

**Reply by MichaelAldrich** [reply]
@sgodanas I put together an example graph to show all 3 of these examples given by lucas.
[functions_between_nodes.nxt.zip](https://github.com/nxt-dev/nxt_editor/files/5851676/functions_between_nodes.nxt.zip)

 There are 3 start points. Run from the farthest left one, it will run with no errors, but run from any other will display the limitations of each pattern. I've included comments in the graph to help demonstrate.

Some of the example code in this graph is in the world node, you can access by double clicking the layer name in the layer manager, or by using the node selector via `/`, which should work in this graph since it already has a world node. 

When forward slash followed by return didn't work for you, you found a bug! I've reported it here: https://github.com/nxt-dev/nxt_editor/issues/148 This problem was specific to your case of not having a world node.

**Reply by sgodanas** [reply]
Thank you folks :) Btw, I am doing a NXT presentation and demo to my colleagues this Friday!!
